test/adapter/json_api/links_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'test_helper'
+
+LinkPost = Class.new(Model)
+class LinkPostSerializer < ActiveModel::Serializer
+  belongs_to :link_author
+end
+
+LinkAuthor = Class.new(Model)
+class LinkAuthorSerializer < ActiveModel::Serializer
+  links do
+    link :self, href: "//link_author/#{object.id}"
+  end
+end
+
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        class LinksTest < Minitest::Test
+          def setup
+            @author = LinkAuthor.new(id: 1337)
+          end
+
+          def test_resource_links
+            hash = ActiveModel::SerializableResource.new(@author, adapter: :json_api).serializable_hash
```

Comment:
if you'd like, there's a shorthand `serializable` in https://github.com/rails-api/active_model_serializers/blob/d02cd30fe55a3ea85e1d351b6e039620903c1871/test/support/serialization_testing.rb#L34-L36